StreamChatClient_v3/Database/DTOs/ChannelDTO.swift

```diff
@@ -95,12 +95,49 @@ extension NSManagedObjectContext {
 extension ChannelDTO {
     static func channelListFetchRequest(query: ChannelListQuery) -> NSFetchRequest<ChannelDTO> {
         let request = NSFetchRequest<ChannelDTO>(entityName: "ChannelDTO")
-        request.sortDescriptors = [.init(key: "cid", ascending: true)] // TODO: sorting from query
+        request.sortDescriptors = query.sort.sortDescriptors
         request.predicate = NSPredicate(format: "ANY queries.filterHash == %@", query.filter.filterHash)
         return request
     }
 }
 
+public extension Array where Element == Sorting {
+    var sortDescriptors: [NSSortDescriptor] {
+        if isEmpty || (count == 1 && first?.field == "updated_at") {
+            return .defaultSortDescriptors(isAscending: first?.isAscending ?? false)
+        }
+        
+        var sortDescriptors = [NSSortDescriptor]()
+        
+        forEach { sorting in
+            var key: String?
+            
+            switch sorting.field {
+            case "cid", "id": key = "cid"
+            case "type", "typeRawValue": key = "typeRawValue"
+            case "created_at", "createdDate": key = "createdDate"
+            case "deleted_at", "deletedDate": key = "deletedDate"
+            case "updated_at", "updatedDate": key = "defaultSortingDate"
+            case "last_message_at", "lastMessageDate": key = "lastMessageDate"
+            default: break
+            }
+            
+            if let key = key {
+                sortDescriptors.append(.init(key: key, ascending: sorting.isAscending))
+            }
+        }
+        
+        return sortDescriptors.isEmpty ? .defaultSortDescriptors(isAscending: false) : sortDescriptors
+    }
```

Comment:
There's awful lot of logic in these couple of lines and it's not tested.

Reply:
I'll add tests.